### ISSUE TITLE
[ENH ]Use sharex=True and sharey=True in some gallery examples

### ISSUE THREAD

**Issue by emmanuelle** [scikit-image/scikit-image#1686]
In a large fraction of gallery examples, we use different subplots to display different filtered versions of an image. For such cases, it would be nice to use the keyword arguments of ``plt.subplots`` ``sharex=True, sharey=True`` so that zooming in one of the subplots also updates the other ones.

**Comment by stefanv** [comment]
:+1:

**Comment by MartinSavc** [comment]
I'd like to look into this one. Are these a specific group of examples or just any example in 'doc/examples/' that would need this enhancement?

**Comment by emmanuelle** [comment]
Thank you @MartinSavc ! 

This enhancement would just be needed for a fraction of the examples of the gallery: those in which all subplots represent an image with the same shape, that correspond usually to different filters applied to the same original image. Such examples include
http://scikit-image.org/docs/dev/auto_examples/plot_denoise.html#example-plot-denoise-py
http://scikit-image.org/docs/dev/auto_examples/plot_ihc_color_separation.html#example-plot-ihc-color-separation-py
http://scikit-image.org/docs/dev/auto_examples/plot_join_segmentations.html#example-plot-join-segmentations-py
For other examples where different kinds of data are plotted in different subpots, such as
http://scikit-image.org/docs/dev/auto_examples/plot_otsu.html
one does not wish that the axes of the different subplots are "tied together".

Hope this is clear, don't hesitate to ask for more advice if you need to.

**Comment by MartinSavc** [comment]
I've updated some of the examples here [example_subplots_share_axes](/MartinSavc/scikit-image/tree/example_subplots_share_axes).

There are a few issues with these. When sharing axes, they seem to change the format of the  containers. The containers usually fill the vertical space of the windows, instead of wrapping the image. This can be hidden by calling axes.set_axes_off(). But the title of each image will still be pushed to the top, as the container didn't actually shrink.

This might be more noticeable on my computer as I use a tiling window manager, so the windows are often forced in a narrow and tall shape.

When using plt.subplots I can only specify If I want axes to be shared in the whole figure, in columns, or in rows. If I use add_subplot, like I did in plot_equalize.py I can explicitly specify the axes object to share with. This may also work across figures, as in applications/plot_coins_segmentation.py. But it would add overhead code which would distract from the example. What are your thoughts on this?

**Comment by emmanuelle** [comment]
@MartinSavc Indeed, I didn't expect that sharing axes would change the dimensions of subaxes. You'll have to modify slightly the value of ``top`` in ``fig.subplots_adjust`` to make it look good again.

I agree with you that sharing axes between figures might be too much magics and too much additional code (but thanks for mentioning that it's possible, I didn't know!)

**Comment by MartinSavc** [comment]
When sharing axes the adjustable axes parameter is set to 'datalim'. Setting adjustable to 'box' is used for images, but is not supported for sharing exes (throws an exception). However I can set it to 'box-forced' to get the desired result. Somewhat confusing. Unfortunately for me, 'box-forced' does not seem to work across figures (only across plots in a single figure).

I've updated the examples I've did so far (here)[https://github.com/MartinSavc/scikit-image/tree/example_subplots_share_axes], which hopefully fixed the problem. There are plenty of different ways to set adjustable, so I've tried to adapt it to each example specifically.

Here are some examples of creating subplots with shared axes. Are any of theses preferred (or undesired)?
```python
#my favorite approach
fig = plt.figure()
ax1 = plt.subplot(1, 2, 1, adjustable='box-forced')
ax2 = plt.subplot(1, 2, 2, sharex=ax1, sharey=ax1, adjustable='box-forced')

#using Figure.add_subplot instead of plt.subplot
ax2 = fig.add_subplot(1, 2, 2, sharex=ax1, sharey=ax1, adjustable='box-forced')

#using subplots, this get very long 
fig,ax = subplots(1, 2, sharex=True, sharey=True, subplot_kw={'adjustable':'box-forced'})

#explicitly for each axes
ax2.set_adjustable('box-forced')
```

**Comment by MartinSavc** [comment]
So, how does it look? Are you still reviewing my submission? Is there anything I should fix, or should I just rebase on upstream?

**Comment by emmanuelle** [comment]
I just answered in #1704. I don't think you need to rebase, Travis and Appveyor are still happy.

**Comment by tacaswell** [comment]
> Setting adjustable to 'box' is used for images, but is not supported for sharing exes (throws an exception)

Sorry to hijack the thread, but I don't think mpl should be raising exceptions in that case.  It might do something non-desirable (but _technically_ correct), but not raise.

**Comment by MartinSavc** [comment]
I pushed my latest commit. I reverted the examples, that did not need the alternative figure creation code, and the ones that did not benefit significantly from axes sharing (using my personal judgment). The following examples kept the alternative syntax:

plot_log_gamma.py - useful to look at details after gamma correction
plot_otsu.py - useful to compare how details are segmented
plot_template.py -useful to look at the exact location of the maximum
plot_equalize.py - useful to look at details after equalization
plot_local_equalize.py - useful to look at details after equalization
plot_register_translation.py - useful to manually inspect translation
plot_ihc_color_separation.py - good example of cross figure axes sharing
plot_adapt_rgb.py - good example of cross figure axes sharing 

I've might have missed some. I've actually missed the last one before commiting. The last two have minimal benefit from axis sharing, but are a good example of cross-figure axes sharing. I would revert only one of them to the preferred figure creation code (the last one).  

I've rebased my local branch which caused a bit of an ugly merge with MartinSavc/scikit-image later on. If this is a problem, I would happily fix it, however I would ask for a how-to or guideline (I'm not quite sure how to fix it at this point).
